pycbc/events/eventmgr_cython.pyx

```diff
@@ -253,3 +253,55 @@ def coincbuffer_numgreater(
     for idx in range(length):
         count += cbuffer[idx] > value
     return count
+
+
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def timecluster_cython(
+    unsigned int[:] indices,
+    long int[:] left,
+    long int[:] right,
+    double[:] stat,
+    int leftlen,
+):
+    cdef:
+        int i, j, k, max_loc
+        long int l, r
+        double max_val
+
+    # i is the index we are inspecting, j is the next one to save
+    i = 0
+    j = 0
+    while i < leftlen:
+        l = left[i]
+        r = right[i]
+
+        # If there are no other points to compare it is obviously the max
+        if (r - l) == 1:
+            indices[j] = i
+            j += 1
+            i += 1
+            continue
+
+        # Find the location of the maximum within the time interval around i
+        # Following block replaces max_loc = argmax(stat[l:r]) + l
+        max_val = -99999999999
+        for k in range(l, r):
```

Comment:
I know in practice this will not be an issue, but this magic large negative constant makes me shiver a bit. Would you be against this, instead:
```suggestion
        max_val = stat[l]
        max_loc = l
        for k in range(l + 1, r):
```

Reply:
That makes sense, and probably helps run time, if (r -l) is small, which it will be in many cases.